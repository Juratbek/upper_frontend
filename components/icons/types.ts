export interface IIconProps {
  color?: string;
  variant?: 'outlined' | 'fulfilled';
  width?: number;
  height?: number;
  opacity?: number;
}
