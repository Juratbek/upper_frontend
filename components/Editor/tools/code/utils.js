export function renderSettings(settings, onClick, context) {
  const wrapper = document.createElement("div");

  settings.forEach((tune) => {
    const button = document.createElement("div");

    button.classList.add(context.api.styles.settingsButton);
    button.classList.add(tune.className);
    button.onclick = () => {
      onClick(tune);
      button.classList.toggle(context.api.styles.settingsButtonActive);
    };
    button.innerHTML = tune.icon;
    wrapper.appendChild(button);
  });

  return wrapper;
}
