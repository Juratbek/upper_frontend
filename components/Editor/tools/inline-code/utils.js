export const isTextEmpty = (textContent) => {
  return /^\s*$/.test(textContent);
};

export const isCaretAtLastPosition = (element) => {
  // Get the selection object
  const selection = window.getSelection();

  if (selection.rangeCount === 0) return false;

  // Get the range object
  const range = selection.getRangeAt(0);

  // Check if the end offset of the range is equal to the length of the text content
  return range.endOffset === element.textContent.length;
};

export const isTextNode = (node) => {
  return node && node.nodeType === 3;
};

export const startsWithCharacters = (text) => {
  if (typeof text !== 'string') return false;
  return /^\S/.test(text);
};
