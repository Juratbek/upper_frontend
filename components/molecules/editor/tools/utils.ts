import { KeyboardEvent } from 'react';

import { IBlockData, IBlockNode } from '../instance/Editor.types';
import { IEditorAPI } from './tool.types';

export const generateBlockNode = (element: HTMLDivElement, block: IBlockData): IBlockNode => {
  return {
    ...element,
    type: block.type,
    id: block.id,
    data: block.data,
  };
};

export const textBlockKeydownHandler = (
  event: KeyboardEvent,
  api: IEditorAPI,
  element: HTMLDivElement,
  block: IBlockData,
) => {
  const { code, metaKey: isMetaKey } = event;

  // if user pressed Enter with ctrl or command add new paragraph
  if (code === 'Enter' && isMetaKey) {
    const node = generateBlockNode(element, block);

    api.addBlock('paragraph', node);
    return;
  }

  // if paragraph is empty when user presses Back key remove this paragraph
  if (code === 'Backspace' && element.innerText === '') {
    api.removeBlock(block.id);
    return;
  }

  if (code === 'ArrowUp') {
    api.focusPreviousText(block.id);
  }
};
