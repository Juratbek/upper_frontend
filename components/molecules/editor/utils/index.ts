import { EDITOR_BLOCKS_CONTAINER_ID } from '../instance/Editor';

export function getEditorBlocksContainer() {
  return document.getElementById(EDITOR_BLOCKS_CONTAINER_ID);
}
