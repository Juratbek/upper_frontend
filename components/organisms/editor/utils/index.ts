import { EDITOR_BLOCKS_CONTAINER_ID, EDITOR_ROOT_ID } from '../instance/Editor';

export function getEditorBlocksContainer() {
  return document.getElementById(EDITOR_BLOCKS_CONTAINER_ID);
}

export function getEditorRoot() {
  return document.getElementById(EDITOR_ROOT_ID);
}
