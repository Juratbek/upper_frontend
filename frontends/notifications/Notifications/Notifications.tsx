import { ApiErrorBoundary, NotificationSkeleton } from 'components';
import { StorysetImage } from 'components/lib';
import { useRouter } from 'next/router';
import { FC, useEffect, useMemo } from 'react';
import {
  useNotificationsList,
  useReadNotification,
  useResetNotificationsCount,
} from 'store/clients/notification';
import { NOTIFICATIONS } from 'variables/notification';

export const Notifications: FC = () => {
  useResetNotificationsCount();
  const {
    query: { page },
  } = useRouter();
  const fetchNotificationsRes = useNotificationsList();
  const { mutate: readNotification } = useReadNotification();

  useEffect(() => {
    document.querySelector('#main')?.scrollTo({ top: 0 });
  }, [page]);

  const notifications = useMemo(() => {
    const { list } = fetchNotificationsRes;
    const notifications = list;
    if (!notifications || notifications.length === 0)
      return (
        <div className='text-center'>
          <StorysetImage
            src='/storyset/notifications.svg'
            width={250}
            height={250}
            storysetUri='internet'
          />
          <p>Bloglarga obuna bo&apos;ling va maqolalar haqida xabarlar oling</p>
          <p className='fs-1'>Hozirda xabarlar mavjud emas</p>
        </div>
      );

    return notifications.map((notification) => {
      const Notification = NOTIFICATIONS[notification.type];

      if (!Notification) return <></>;
      return (
        <div key={notification.id} onClick={() => readNotification(notification.id)}>
          <Notification {...notification} />
        </div>
      );
    });
  }, [fetchNotificationsRes.data]);

  return (
    <ApiErrorBoundary
      res={fetchNotificationsRes}
      fallback={<NotificationSkeleton className='px-3 py-2' />}
      fallbackItemCount={3}
      className='tab pt-1'
    >
      {notifications}
    </ApiErrorBoundary>
  );
};
