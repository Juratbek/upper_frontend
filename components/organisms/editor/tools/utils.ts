import { KeyboardEvent } from 'react';

import { IEditorAPI, IEditorContext } from '../context/EditorContext.types';
import { IBlockData, IBlockNode, TInitialBlockData } from '../instance/Editor.types';
import { isEmpty } from '../utils/html';
import { Selection } from '../utils/selection';

export const generateBlockNode = (element: HTMLDivElement, block: IBlockData): IBlockNode => {
  // just creating new object from params will not work
  // because element is provided using react ref and stores reference to the reference node not to the actual element
  return Object.assign(element, {
    type: block.type,
    id: block.id,
    data: block.data,
  } satisfies IBlockData);
};

export const textBlockKeydownHandler = <T extends { text: string }>(
  event: KeyboardEvent,
  api: IEditorAPI,
  element: HTMLElement,
  blockId: IBlockData['id'],
  config?: { shouldMergeOnBackspace?: boolean },
) => {
  const { code } = event;

  // if user pressed Enter with ctrl or command add new paragraph
  if (code === 'Enter') {
    api.addBlock('paragraph', blockId);
    event.preventDefault();
    return;
  }

  // if paragraph is empty when user presses Back key remove this paragraph
  if (code === 'Backspace' && isEmpty(element)) {
    api
      .removeBlock(blockId)
      .then(({ prevBlocks }) => focusPreviousBlock(prevBlocks, blockId, event));
    return;
  }

  if (code === 'Backspace' && config?.shouldMergeOnBackspace) {
    const selection = Selection.selection;
    if (!selection) return;

    const range = selection.getRangeAt(0);
    const offset = range.startOffset;

    if (offset === 0) {
      api.mergeWithPrevBlock<T>(blockId, (prevBlock, currentBlock) => {
        prevBlock.data.text += ` ${currentBlock.data.text}`;

        return prevBlock.data;
      });
      return;
    }
  }
};

export function getCurrentBlock<T>({ data, hoveredBlock }: IEditorContext<T>) {
  return data.find((b) => hoveredBlock?.id === b.id);
}

function focusPreviousBlock(
  prevBlocks: IBlockData<TInitialBlockData>[],
  blockId: IBlockData['id'],
  event: KeyboardEvent,
) {
  const prevBlock = prevBlocks.find((_, index) => {
    const nextBlock = prevBlocks[index + 1];
    // if next block is the deleted block -> current block is previous block
    if (nextBlock?.id === blockId) return true;

    return false;
  });

  if (!prevBlock) return;

  const prevBlockElement = document.getElementById(prevBlock?.id);
  const textElements = prevBlockElement?.querySelectorAll(
    'h1, h2, h3, h4, h5, h6, p',
  ) as NodeListOf<HTMLParagraphElement | HTMLHeadingElement>;

  if (!textElements || textElements.length < 1) return;

  const lastTextElement = textElements[textElements.length - 1];
  lastTextElement.focus();

  // Create a range (a span of content) and a selection (current user selection)
  const range = document.createRange();
  const selection = window.getSelection();

  if (!selection) return;

  // Move the range to the end of the content inside the element
  range.selectNodeContents(lastTextElement);
  range.collapse(false);

  // Remove any current selections
  selection.removeAllRanges();

  // Add the new range to the selection
  selection.addRange(range);

  event.preventDefault();
}
