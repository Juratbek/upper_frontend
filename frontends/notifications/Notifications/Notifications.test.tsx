import { render } from '@testing-library/react';
import { QueryWrapper, ReduxWrapper } from 'tests';
import { INotification } from 'types';
import { Override } from 'utils';
import { vi } from 'vitest';

import { Notifications } from './Notifications';

const mockArticle: INotification['article'] = {
  id: 1,
  title: 'Vitest mocking',
};

const mockAuthor: INotification['author'] = {
  id: 12,
  imgUrl: 'https://image.upper.uz/12',
  name: 'Aziz',
};

const mockNotifications: INotification[] = [
  {
    article: mockArticle,
    author: mockAuthor,
    createdDate: '12.10.2024',
    id: 10,
    status: 'UNREAD',
    type: 'ARTICLE_PUBLISHED',
  },
  {
    article: mockArticle,
    author: mockAuthor,
    createdDate: '12.10.2024',
    id: 11,
    status: 'UNREAD',
    type: 'ARTICLE_PUBLISHED',
  },
  {
    article: mockArticle,
    author: mockAuthor,
    createdDate: '12.10.2024',
    id: 12,
    status: 'READ',
    type: 'COMMENT',
  },
  {
    article: mockArticle,
    author: mockAuthor,
    createdDate: '12.10.2024',
    id: 13,
    status: 'UNREAD',
    type: 'LIKED',
  },
];

const mocks = vi.hoisted(() => ({
  useNotificationsList: vi.fn().mockReturnValue({ list: [], isPending: true, isLoading: true }),
}));

vi.mock('store/clients/notification', async (importOriginal) => {
  const original = await importOriginal<typeof import('store/clients/notification')>();

  return {
    ...original,
    useNotificationsList: mocks.useNotificationsList,
  };
});

const Component = (
  <ReduxWrapper>
    <QueryWrapper>
      <Notifications />
    </QueryWrapper>
  </ReduxWrapper>
);

describe('Notifications page', () => {
  it('snapshot', () => {
    render(Component);
    expect(document.body).toMatchSnapshot();
  });

  it('renders notifications data from api', () => {
    mocks.useNotificationsList.mockReturnValue({
      isPending: false,
      isLoading: false,
      isSuccess: true,
      list: mockNotifications,
    });
    render(Component);

    expect(document.body).toMatchSnapshot();
  });

  it('renders with Fetch Next Page button', () => {
    mocks.useNotificationsList.mockReturnValue({
      isPending: false,
      isLoading: false,
      isSuccess: true,
      hasNextPage: true,
      list: mockNotifications,
    });
    render(Component);

    expect(document.body).toMatchSnapshot();
  });

  it('renders empty list UI', () => {
    mocks.useNotificationsList.mockReturnValue({
      isPending: false,
      isLoading: false,
      isSuccess: true,
      list: [],
    });
    render(Component);

    expect(document.body).toMatchSnapshot();
  });

  it('renders nothing if Notification component for notification type is not found', () => {
    const mockNotification: Override<INotification, { type: string }> = {
      article: mockArticle,
      author: mockAuthor,
      createdDate: '12.10.2024',
      id: 13,
      status: 'UNREAD',
      type: 'NOT_SUPPORTED',
    };

    mocks.useNotificationsList.mockReturnValue({
      isPending: false,
      isLoading: false,
      isSuccess: true,
      hasNextPage: false,
      list: [mockNotification],
    });
    render(Component);

    expect(document.body).toMatchSnapshot();
  });
});
