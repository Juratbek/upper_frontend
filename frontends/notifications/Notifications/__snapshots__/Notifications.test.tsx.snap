// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Notifications page > renders empty list UI 1`] = `
<body>
  <div>
    <div
      class="tab"
    >
      <div
        class="text-center"
      >
        <span>
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height="250"
            loading="lazy"
            src="/storyset/notifications.svg"
            style="color: transparent;"
            width="250"
          />
          <br />
          <a
            class="storyset-link link"
            href="https://storyset.com/internet"
            rel="noreferrer"
            target="_blank"
          >
            Data illustrations by Storyset
          </a>
        </span>
        <p>
          Bloglarga obuna bo'ling va maqolalar haqida xabarlar oling
        </p>
        <p
          class="fs-1"
        >
          Hozirda xabarlar mavjud emas
        </p>
      </div>
    </div>
  </div>
</body>
`;

exports[`Notifications page > renders nothing if Notification component for notification type is not found 1`] = `
<body>
  <div>
    <div
      class="tab"
    />
  </div>
</body>
`;

exports[`Notifications page > renders notifications data from api 1`] = `
<body>
  <div>
    <div
      class="tab"
    >
      <div>
        <a
          class="root unread link"
          href="/web/articles/1"
        >
          <h3
            class="title"
          >
            Vitest mocking
          </h3>
          <div
            class="footer align-items-center"
          >
            <div
              class="author"
            >
              <div
                class="avatar avatar--small"
                style="border-color: rgba(245, 245, 245, 0.1);"
              >
                <img
                  alt="UPPER"
                  data-nimg="fill"
                  decoding="async"
                  loading="lazy"
                  sizes="100vw"
                  src="http://localhost:3000/_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=3840&q=75 3840w"
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
                />
              </div>
              <span
                class="author-name"
              >
                Aziz
              </span>
            </div>
            <span
              class="date"
            >
              Hozirgina
            </span>
          </div>
        </a>
      </div>
      <div>
        <a
          class="root unread link"
          href="/web/articles/1"
        >
          <h3
            class="title"
          >
            Vitest mocking
          </h3>
          <div
            class="footer align-items-center"
          >
            <div
              class="author"
            >
              <div
                class="avatar avatar--small"
                style="border-color: rgba(245, 245, 245, 0.1);"
              >
                <img
                  alt="UPPER"
                  data-nimg="fill"
                  decoding="async"
                  loading="lazy"
                  sizes="100vw"
                  src="http://localhost:3000/_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=3840&q=75 3840w"
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
                />
              </div>
              <span
                class="author-name"
              >
                Aziz
              </span>
            </div>
            <span
              class="date"
            >
              Hozirgina
            </span>
          </div>
        </a>
      </div>
      <div>
        <a
          class="root link"
          href="/web/articles/1"
        >
          <h3
            class="title"
          >
            Aziz "Vitest mocking" maqolangizga fikr bildirdi
          </h3>
          <div
            class="footer"
          >
            <div
              class="d-flex align-items-center"
            >
              <div
                class="icon-box"
              >
                <svg
                  fill="none"
                  height="24"
                  role="icon"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M2 19.7307V6.05541C2 4.82811 2.99492 3.83319 4.22222 3.83319H19.7778C21.0051 3.83319 22 4.82811 22 6.05541V15.51C22 16.7373 21.0051 17.7322 19.7778 17.7322H7.5125C6.83742 17.7322 5.77723 18.5662 5.77723 18.5662L3.5 19.9999C3.5 19.9999 2 20.5163 2 19.7307Z"
                    fill="#007AFF"
                  />
                  <path
                    d="M6.5 13.0915H17.5"
                    stroke="#E6F2FF"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                  <path
                    d="M6.50101 9.09153L15.501 9.09149"
                    stroke="#E6F2FF"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </div>
              <p
                class="comment-text"
              />
            </div>
            <time
              class="date mt-auto"
            >
              Hozirgina
            </time>
          </div>
        </a>
      </div>
      <div>
        <a
          class="root unread link"
          href="/web/articles/1"
        >
          <h3
            class="title"
          >
            "Vitest mocking" maqolangiz Azizga yoqdi
          </h3>
          <div
            class="footer"
          >
            <div
              class="icon-box"
            >
              <svg
                fill="none"
                height="24"
                role="icon"
                viewBox="0 0 24 24"
                width="24"
              >
                <path
                  d="M14 9V5C14 5 14 2 11 2L7 11V22H18.28C18.7623 22.0055 19.2304 21.8364 19.5979 21.524C19.9654 21.2116 20.2077 20.7769 20.28 20.3L21.66 11.3C21.7035 11.0134 21.6842 10.7207 21.6033 10.4423C21.5225 10.1638 21.3821 9.90629 21.1919 9.68751C21.0016 9.46873 20.7661 9.29393 20.5016 9.17522C20.2371 9.0565 19.9499 8.99672 19.66 9H14ZM7 22H4C3.46957 22 2.96086 21.7893 2.58579 21.4142C2.21071 21.0391 2 20.5304 2 20V13C2 12.4696 2.21071 11.9609 2.58579 11.5858C2.96086 11.2107 3.46957 11 4 11H7"
                  fill="#007AFF"
                />
                <path
                  d="M7 11L11 2C14 2 14 5 14 5V9H19.66C19.9499 8.99672 20.2371 9.0565 20.5016 9.17522C20.7661 9.29393 21.0016 9.46873 21.1919 9.68751C21.3821 9.90629 21.5225 10.1638 21.6033 10.4423C21.6842 10.7207 21.7035 11.0134 21.66 11.3L20.28 20.3C20.2077 20.7769 19.9654 21.2116 19.5979 21.524C19.2304 21.8364 18.7623 22.0055 18.28 22H7M7 11V22M7 11H4C3.46957 11 2.96086 11.2107 2.58579 11.5858C2.21071 11.9609 2 12.4696 2 13V20C2 20.5304 2.21071 21.0391 2.58579 21.4142C2.96086 21.7893 3.46957 22 4 22H7"
                  stroke="#E6F2FF"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </div>
            <time
              class="date mt-auto"
            >
              Hozirgina
            </time>
          </div>
        </a>
      </div>
    </div>
  </div>
</body>
`;

exports[`Notifications page > renders with Fetch Next Page button 1`] = `
<body>
  <div>
    <div
      class="tab"
    >
      <div>
        <a
          class="root unread link"
          href="/web/articles/1"
        >
          <h3
            class="title"
          >
            Vitest mocking
          </h3>
          <div
            class="footer align-items-center"
          >
            <div
              class="author"
            >
              <div
                class="avatar avatar--small"
                style="border-color: rgba(245, 245, 245, 0.1);"
              >
                <img
                  alt="UPPER"
                  data-nimg="fill"
                  decoding="async"
                  loading="lazy"
                  sizes="100vw"
                  src="http://localhost:3000/_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=3840&q=75 3840w"
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
                />
              </div>
              <span
                class="author-name"
              >
                Aziz
              </span>
            </div>
            <span
              class="date"
            >
              Hozirgina
            </span>
          </div>
        </a>
      </div>
      <div>
        <a
          class="root unread link"
          href="/web/articles/1"
        >
          <h3
            class="title"
          >
            Vitest mocking
          </h3>
          <div
            class="footer align-items-center"
          >
            <div
              class="author"
            >
              <div
                class="avatar avatar--small"
                style="border-color: rgba(245, 245, 245, 0.1);"
              >
                <img
                  alt="UPPER"
                  data-nimg="fill"
                  decoding="async"
                  loading="lazy"
                  sizes="100vw"
                  src="http://localhost:3000/_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimage.upper.uz%2F12&w=3840&q=75 3840w"
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
                />
              </div>
              <span
                class="author-name"
              >
                Aziz
              </span>
            </div>
            <span
              class="date"
            >
              Hozirgina
            </span>
          </div>
        </a>
      </div>
      <div>
        <a
          class="root link"
          href="/web/articles/1"
        >
          <h3
            class="title"
          >
            Aziz "Vitest mocking" maqolangizga fikr bildirdi
          </h3>
          <div
            class="footer"
          >
            <div
              class="d-flex align-items-center"
            >
              <div
                class="icon-box"
              >
                <svg
                  fill="none"
                  height="24"
                  role="icon"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M2 19.7307V6.05541C2 4.82811 2.99492 3.83319 4.22222 3.83319H19.7778C21.0051 3.83319 22 4.82811 22 6.05541V15.51C22 16.7373 21.0051 17.7322 19.7778 17.7322H7.5125C6.83742 17.7322 5.77723 18.5662 5.77723 18.5662L3.5 19.9999C3.5 19.9999 2 20.5163 2 19.7307Z"
                    fill="#007AFF"
                  />
                  <path
                    d="M6.5 13.0915H17.5"
                    stroke="#E6F2FF"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                  <path
                    d="M6.50101 9.09153L15.501 9.09149"
                    stroke="#E6F2FF"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </div>
              <p
                class="comment-text"
              />
            </div>
            <time
              class="date mt-auto"
            >
              Hozirgina
            </time>
          </div>
        </a>
      </div>
      <div>
        <a
          class="root unread link"
          href="/web/articles/1"
        >
          <h3
            class="title"
          >
            "Vitest mocking" maqolangiz Azizga yoqdi
          </h3>
          <div
            class="footer"
          >
            <div
              class="icon-box"
            >
              <svg
                fill="none"
                height="24"
                role="icon"
                viewBox="0 0 24 24"
                width="24"
              >
                <path
                  d="M14 9V5C14 5 14 2 11 2L7 11V22H18.28C18.7623 22.0055 19.2304 21.8364 19.5979 21.524C19.9654 21.2116 20.2077 20.7769 20.28 20.3L21.66 11.3C21.7035 11.0134 21.6842 10.7207 21.6033 10.4423C21.5225 10.1638 21.3821 9.90629 21.1919 9.68751C21.0016 9.46873 20.7661 9.29393 20.5016 9.17522C20.2371 9.0565 19.9499 8.99672 19.66 9H14ZM7 22H4C3.46957 22 2.96086 21.7893 2.58579 21.4142C2.21071 21.0391 2 20.5304 2 20V13C2 12.4696 2.21071 11.9609 2.58579 11.5858C2.96086 11.2107 3.46957 11 4 11H7"
                  fill="#007AFF"
                />
                <path
                  d="M7 11L11 2C14 2 14 5 14 5V9H19.66C19.9499 8.99672 20.2371 9.0565 20.5016 9.17522C20.7661 9.29393 21.0016 9.46873 21.1919 9.68751C21.3821 9.90629 21.5225 10.1638 21.6033 10.4423C21.6842 10.7207 21.7035 11.0134 21.66 11.3L20.28 20.3C20.2077 20.7769 19.9654 21.2116 19.5979 21.524C19.2304 21.8364 18.7623 22.0055 18.28 22H7M7 11V22M7 11H4C3.46957 11 2.96086 11.2107 2.58579 11.5858C2.21071 11.9609 2 12.4696 2 13V20C2 20.5304 2.21071 21.0391 2.58579 21.4142C2.96086 21.7893 3.46957 22 4 22H7"
                  stroke="#E6F2FF"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </div>
            <time
              class="date mt-auto"
            >
              Hozirgina
            </time>
          </div>
        </a>
      </div>
      <button
        class="button mt-2"
        color="tertiary"
        style="opacity: 1; font-size: 20px; width: 100%;"
      >
        Yana yuklash
      </button>
    </div>
  </div>
</body>
`;

exports[`Notifications page > snapshot 1`] = `
<body>
  <div>
    <div
      class="tab"
    >
      <div
        class="content-center"
        style="height: 15rem;"
      >
        <div
          class="spinner spinner--primary"
          role="spinner"
        />
      </div>
    </div>
  </div>
</body>
`;
